replace turtle with location arrow glyph.

davidsbailey opened the pull request:
screencast: http://screencast.com/t/wcNMSnZJ

davidsbailey commented on apps/src/applab/applab.js:
```diff
@@ -1747,9 +1747,11 @@ Applab.imageUploadButton = function (opts) {
-// its x,y coordinates. The image is currently 31x45, so these offsets are 50%
-var TURTLE_X_OFFSET = -15;
-var TURTLE_Y_OFFSET = -22;
+// its x,y coordinates. The image is currently 44x44, with the center at
```
an alternate image is available at 66x66, with the center at  (36,30), which could be rendered at a retina-friendly size of 33x33.

cpirich commented:
Seems way more complicated than if we just had an image padded with transparency so the center was still in the middle. Can we ask Ross to do that? Also, shouldn't the retina image be 2x the size?

davidsbailey commented:
Some of the complication (msTransform, webkitTransform) is needed to make the turtle rotate at all in other browsers, however, it is true the code would be simpler with a centered image.

This image is already a retina-friendly 2x the size of the space its rendered in.

davidsbailey commented:
I've resized the image to avoid setting transformOrigin.

Bjvanminnen commented on apps/src/applab/applab.js:
```diff
@@ -1747,9 +1747,10 @@ Applab.imageUploadButton = function (opts) {
+// its x,y coordinates. The image is currently 48x48, rendered at 24x24.
+var TURTLE_X_OFFSET = -12;
+var TURTLE_Y_OFFSET = -12;
+var TURTLE_ROTATION_OFFSET = -45;
```
Would it make more sense to just rotate the image that we check in by 45 degrees instead of having this offset in code?

I might also try to make the x/y offset a little less mysterious, either setting them to "-24 / 2" (a little more clear why this number) or by creating a separate TURTLE_WIDTH/TURTLE_HEIGHT and setting to -TURTLE_WIDTH / 2

Bjvanminnen commented on apps/src/applab/applab.js:
```diff
@@ -1779,7 +1780,11 @@ function updateTurtleImage(turtleImage) {
+  var transform = 'rotate(' + heading + 'deg)';
+  turtleImage.style.transform = transform;
+  turtleImage.style.msTransform = transform;
+  turtleImage.style.webkitTransform = transform;
```
What's the reason for having the turtle be an img element on top of the canvas instead of drawing the image onto the canvas?

davidsbailey commented:
Chris: Rotating the image, saving it, and then rotating it again makes the image look somewhat more pixelated:

![screen shot 2015-04-15 at 9 55 33 am](https://cloud.githubusercontent.com/assets/8001765/7164531/a3e913e0-e355-11e4-8301-db879e80ed36.png)

(ignore that the nose is chopped off, that's because I didn't bother resizing the canvas when I rotated)

Brent: Chris made the decision to use a separate HTML element, but presumably the reason is that a separate HTML element eliminates the need to later remove the image from the canvas, which is hard because the canvas doesn't have any memory of what was there before once it has been painted to: http://stackoverflow.com/questions/7576755/how-to-undraw-hide-remove-or-delete-an-image-from-an-html-canvas

davidsbailey commented:
Merging this now, having addressed all comments, since Baker needs this in order to record videos. If there is more feedback I will address it in a follow-up PR.
